src/main/java/com/palantir/gradle/revapi/RevapiPlugin.java

```diff
@@ -60,14 +61,40 @@ public void apply(Project project) {
                     Configuration revapiNewApi = project.getConfigurations().create("revapiNewApi", conf -> {
                         conf.extendsFrom(
                                 project.getConfigurations().getByName(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME));
+                        conf.setCanBeConsumed(false);
                     });
 
-                    task.dependsOn(allJarTasksIncludingDependencies(project, revapiNewApi));
+                    Configuration revapiNewApiElements = project.getConfigurations()
+                            .create("revapiNewApiElements", conf -> {
+                                conf.extendsFrom(project.getConfigurations()
+                                        .getByName(JavaPlugin.API_ELEMENTS_CONFIGURATION_NAME));
+                                conf.setCanBeConsumed(false);
+                            });
```

Comment:
this will resolve the runtime variant of transitives, so you probably want to set a usage attribute on here, essentially copying what attributes compileClasspath sets:
https://github.com/gradle/gradle/blob/193008197a7b450fd57970f8598ca3f1ffc810b7/subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JvmPluginsHelper.java#L224-L230

some of these were not always public / change with the gradle version though, so I'd almost recommend just stealing the attributes from compileClasspath and setting them one by one

Reply:
and do the same for `revapiNewApi`